### ISSUE TITLE
Steam client 2019-01-16 needs ipv6 kernel module to reach ipv4 cdn

### ISSUE THREAD

**Issue by tomaszg7** [ValveSoftware/steam-for-linux#6028]
#### Your system information

* Steam client version (build number or date): Jan 17 2019
* Distribution (e.g. Ubuntu): Gentoo
* Opted into Steam client beta?: yes
* Have you checked for system updates?: yes

#### Please describe your issue in as much detail as possible:
Last update was supposed to fix the problem with 0 bytes updates for many games. In my case it only created a bigger problem. I still see the needed 0 bytes updates in *library* view, but no entries appear in *download* view. When I start the game I get a popup: "An error occurred while updating XXX (content servers unreachable). The issue appeared with previous update, today's fix didn't help. Both Steam Play and native games seem to be affected.

Moreover I can't install new games with exactly the same error message. I see things like
```
GameAction [AppID 740090, ActionID 9] : InstallApps failed with AppError_21 with ""
GameAction [AppID 740090, ActionID 10] : LaunchApp changed task to DownloadingDepots with ""
GameAction [AppID 740090, ActionID 10] : LaunchApp failed with AppError_21 with ""
GameAction [AppID 740090, ActionID 10] : LaunchApp changed task to Failed with ""
```
in console. Checking game files stalls at 0%

I'd like to add that I had problems getting the client update to run. After automatic update the client wouldn't start (silently, or after some fiddling complaining about missing `run.sh` in steam-runtime). I had to delete `.steam`, so that client would do a more complete reinstall.

Opting out of beta fixed the issue for me.

#### Steps for reproducing this issue:

1. Try to start a game with "Update queued" label or install a new game.

**Comment by DPTJKKVH** [comment]
Oh I might have created a dupe of your issue here #6033.
Please take a look at my issue as well @tomaszg7, especially to the possible workaround with IPv6. I have a hunch that VALVE might have by accident left in a force IPv6 flag or removed IPv4 compatibility flag for content servers in the beta builds.

@Plagman

**Comment by kisak-valve** [comment]
@tomaszg7, can you check if you have ipv6 enabled in your kernel and if not, if modprobing the ipv6 kernel module (or making it built-in) makes a difference.

**Comment by Youngsie1997** [comment]
Can confirm in my case at least re-enabling ipv6 in systemd-boot "fixes" this issue less than ideal of course.

**Comment by tomaszg7** [comment]
@DPTJKKVH I didn't force enable any Steam Play game. In fact the problem appeared in the update before the one which allowed that.

@kisak-valve I don't have ipv6 enabled nor modules built. I'll recompile the kernel and check again.

**Comment by DPTJKKVH** [comment]
I'll cross-post this from my old issue so others can understand why we are looking into IPv6:

> Some folks over at GamingOnLinux say that this issue happens only when IPv6 is not available and can be fixed by enabling IPv6. See [here](https://www.gamingonlinux.com/articles/another-steam-client-beta-is-out-adds-the-ability-to-force-steam-play.13376/page=7#r146104) and [here](https://www.gamingonlinux.com/articles/another-steam-client-beta-is-out-adds-the-ability-to-force-steam-play.13376/page=8#r146109).

**Comment by DPTJKKVH** [comment]
@tomaszg7 Well it seems I was half right. It might be initially triggered by force enabling SteamPlay but then it applies to all downloads. I will try to confirm this theory and report back.

**Comment by tomaszg7** [comment]
Not sure it applies to all. Right now I reached the state when 0 sized downloads go through, but a real update to a Linux title won't go through :/

**Comment by tomaszg7** [comment]
@kisak-valve Installing ipv6 module seems to fix the error for me, although I don't seem to have ipv6 connectivity from my ISP.

**Comment by DPTJKKVH** [comment]
Some side note: When switching back to STEAM stable build my downloads go haywire. A Windows only game "VALKYRIA CHRONICLES" triggered a complete re download of the game even though I did not change any SteamPlay settings on this one. So I guess there are more issues regarding downloads to be discovered in the current beta client.

This is just a side note for @kisak-valve though because I do not want to derail this issues topic.

**Comment by kisak-valve** [comment]
Hello @DPTJKKVH, please open a new issue report, ideally with a reliable way to reproduce that issue.

**Comment by DPTJKKVH** [comment]
@tomaszg7 I cannot confirm your observation. After downloading 18 GB with the stable client I went back to the beta build and tried to download a game but got the content server unreachable error.

So it is definitely the beta client depending on IPv6 for some reason.

**Comment by lucifertdark** [comment]
My ISP has done nothing in the way of support for IPv6 & isn't planning to do anything in the near future.

**Comment by OlliC** [comment]
Had the same problem. My ISP has ipv6 connecitivity, but i disabled it with `net.ipv6.conf.all.disable_ipv6 = 1` to prevent ipv6 leaks with my vpn. Steam wouldn't download anything saying it cannot reach the download servers until i reenabled ipv6.

**Comment by c1a0** [comment]
Same issue here, ipv6 was previously disabled in my kernel. Requires opting out of the Steam beta client to be able to download any games (old/new installs).

**Comment by badbytes** [comment]
Opted out of beta to get working as well.

**Comment by sdelafond** [comment]
Opting out of Steam beta doesn't change anything for me: updating only succeeds if I set net.ipv6.conf.all.disable_ipv6=0.

**Comment by Johnanater** [comment]
Just updated to beta and also experiencing this. I also have IPv6 disabled.

**Comment by GarlandKey** [comment]
I have IPv6 disabled and can confirm that this is a serious issue. I can't play, update or install anything - steam native or using Steam Play. I went so far as to completely uninstall and delete my steam folder and I still have the same issue. Beta or not, the problem persists.

If you need any information on my system, logs, etc, pls ask and I will provide.

**Comment by rantecki** [comment]
Same issue here.  Opting out fixed it for me.  With IPV6 enabled on the beta I was getting about 4KB/s download.  Now with IPV6 disabled I'm getting 1.5+MB/s.

**Comment by baconsalad** [comment]
Hit by the same issue. Gentoo with custom kernel without IPV6, same error. Updated the kernel and enabled IPV6 as a module, works fine. Different computer with IPV6 compiled into the kernel, never had a problem.

**Comment by ewe2** [comment]
The latest steam client update (Jan 31, 2019 client, package version 1548979876) out today now reproduces this same error. Games and Proton 3.16 beta want updates and content servers unreachable. Completely broken, well done all.

**Comment by Cuddlypuff** [comment]
can confirm that it looks like the official build released yesterday does have the same ipv6-related issues that I had experienced on the beta before opting out. First, with my vpn blocking ipv6 i saw the following in content_log.txt:
```
[2019-02-01 18:24:09] Failed to get "/serverlist/20/20/" from valve401.steamcontent.com (HTTP Error 0)
[2019-02-01 18:24:09] Failed to get "/serverlist/20/20/" from valve400.steamcontent.com (HTTP Error 0)
[2019-02-01 18:24:09] Failed to get "/serverlist/20/20/" from valve803.steamcontent.com (HTTP Error 0)
[2019-02-01 18:24:09] Failed to get list of download sources from any known CS!
[2019-02-01 18:24:09] AppID 305050 update canceled : No servers found for depot 305053, manifest 5295788599734531361 (No connection to content servers)
```
(note that when I had that error on the beta I was able to verify that I could indeed access the steamcontent serverlists through chrome, so it wasn't an issue with my ISP or my connection in general)

Then after allowing ipv6 through my vpn, games still failed to download (although one of them would get in 'starting' state and get stuck there), with the log showing the following:
```
[2019-02-01 18:24:57] Created download interface of type 'CS' (1) to host valve176.steamcontent.com (valve176.steamcontent.com)
[2019-02-01 18:25:17] HTTP (CS,614) - valve614.steamcontent.com (0.0.0.0:0 / 0.0.0.0:0, host: valve614.steamcontent.com): AuthenticateDepotID (305053) - ERROR: Failed to get CDN auth token from Steam
[2019-02-01 18:25:17] HTTP (CS,614) - valve614.steamcontent.com (0.0.0.0:0 / 0.0.0.0:0, host: valve614.steamcontent.com): Closing connection
```
I had to leave so I left it running and I see in the log that I have about 20 minutes of 'failed to get CDN auth token' messages, until it finally went through:
```
[2019-02-01 18:44:02] HTTP (CS,604) - valve604.steamcontent.com (162.254.193.14:80 / 162.254.193.14:80, host: valve604.steamcontent.com): AuthenticateDepotID (404730) - Success!
[2019-02-01 18:44:02] HTTP (CS,604) - valve604.steamcontent.com (162.254.193.14:80 / 162.254.193.14:80, host: valve604.steamcontent.com): valve604.steamcontent.com/depot/404730/manifest/3921972306181039640/5?l=14&e=1549669442&sid=10491621&h=1d2e193c3389d6a19c6ebfe469551ef36aa9d52a - received 200 (OK) HTTP response
```
I'll give it a shot again tomorrow, but I suspect that if I block ipv6 through my vpn again it'll make it impossible to download anything through steam again...

**Comment by trialism** [comment]
My ISP doesn't support ipv6 so fixed this ASAP! I can't play games because of this bug.

**Comment by tomaszg7** [comment]
@trialism As described here, you don't need ipv6 connectivity or any kind of support from ISP. You just need kernel module to be present.

**Comment by trialism** [comment]
@tomaszg7 it doesn't matter and it's supposed to be on in ubuntu1. `test -f /proc/net/if_inet6 && echo "Running kernel is IPv6 ready"` says it's ok. I've an ipv6 address on my wifi device (checked with ifconfig).

**Comment by tomaszg7** [comment]
Well, I don't have any support from ISP, but it works for me. Note that my interface was assigned IPv6 address by my router, maybe that's the important bit, I don't know.

**Comment by trialism** [comment]
@tomaszg7 it doesn't work for me and my ipv6 address was assigned by my router too:

```
inet6 fe80:: ... prefixlen 64  scopeid 0x20<link>
```

What are the steam developers even doing? They should have fixed this in the beta a long time ago.

**Comment by Jasper-Ben** [comment]
> ```
> inet6 fe80:: ... prefixlen 64  scopeid 0x20<link>
> ```
That's your loopback IPv6, like 127.0.0.1 with IPv4, not assigned by router.

**Comment by Basiani** [comment]
In my country only two internet providers are and none of them have support for ipv6. Now I can't use steam anymore. It just not runs. I tried to remove it completly and install on my Arch Linux, but non success. There is no points to go through Windows. So, steam on Linux died in my country.

**Comment by trialism** [comment]
My issue has been fixed in the latest beta client update. @Basiani try to switch to steam beta and let's see if it works for you.

**Comment by c1a0** [comment]
Yes it seems to be working fine for me in the newest beta client, still an issue in stable though. Glad to see a fix coming in!

**Comment by Johnanater** [comment]
Seems to be working for me on the stable branch now.

**Comment by DPTJKKVH** [comment]
@kisak-valve On Steam-Beta it is fixed.

If you tell me if the patch was included in the latest stable release I can test this as well.

**Comment by Youngsie1997** [comment]
Can also confirm working again on the latest beta. ipv6 disabled once again.

**Comment by telans** [comment]
Can also confirm, working on beta ipv4 only.

**Comment by tomaszg7** [comment]
I can also confirm that it seems to be fixed on current beta.

**Comment by kisak-valve** [comment]
Closing as fixed.
